### PR TITLE
Add sourcemapBaseUrl option

### PR DESCRIPTION
**Pull request by nickgarlis**
This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [x] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

### Description

This feature can be useful for when sourcemaps need to be loaded through a different URL to the one where the bundled code is loaded from.

It would be the equivalent of the `publicPath` option in [Webpack's source-map-dev-tool plugin](https://webpack.js.org/plugins/source-map-dev-tool-plugin/)

The use case would be to source my organisation's sourcemaps in a server within our internal network to prevent exposing the source code on production.

If this is something of interest then the following features could also be considered:
- Providing a base URL option for bundle sources (`sources` field inside of *.map files)
- Providing a different output directory for sourcemaps.

### CODE REVIEW

File: src/utils/url.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export function isValidUrl(url: string): boolean {
```
Comment (nickgarlis) [inline src/utils/url.ts:1]: I am not sure where I should add tests for functions like these. It looks like all of the tests in the project are end-to-end tests and do not import typescript code from `src`. @lukastaegert I'd be happy to open a separate PR adding unit tests that execute with `mocha -r ts-node/register`.

Comment (lukastaegert) [reply src/utils/url.ts:1]: No, please don't. Then we would still need a test that actually uses the option, i.e. an E2E test. I mean that is the point, to test the effect of this option. As e2e tests in Rollup are REALLY fast, there is really no point in writing unit tests that need to be adapted with every refactoring.
A very simple way could be to add a chunking-form test because those tests just snapshot the output directory and that would include the source map. Or you add a sourcemap test, those tests pass you the source map and you make assertions on the generated map. If all else fails, I will need to look into writing a test myself.

**Comment by lukastaegert** [comment]
I took the liberty to add a test for the bad case, which also prompted me to improve the error message a little. I also added some documentation for the option itself and shortened the CLI printout to match the 80 character line-break we use in that file. Otherwise, this is really high quality work and I plan to merge it once everything is green.

**Review by lukastaegert — APPROVED** [review]
